### PR TITLE
Gromacs CUDA patch + some version constraints

### PR DESCRIPTION
**Pull request by w8jcik**
- Gromacs CUDA patch
- hwloc and cmake constraints
- Latest Gromacs 4

### CODE REVIEW

File: var/spack/repos/builtin/packages/gromacs/package.py [hunk 1]
```diff
@@ -54,6 +54,7 @@ class Gromacs(CMakePackage):
     version('5.1.5',  sha256='c25266abf07690ecad16ed3996899b1d489cbb1ef733a1befb3b5c75c91a703e')
     version('5.1.4',  sha256='0f3793d8f1f0be747cf9ebb0b588fb2b2b5dc5acc32c3046a7bee2d2c03437bc')
     version('5.1.2',  sha256='39d6f1d7ae8ba38cea6089da40676bfa4049a49903d21551abc030992a58f304')
+    version('4.6.7', sha256='6afb1837e363192043de34b188ca3cf83db6bd189601f2001a1fc5b0b2a214d9')
```
Comment (tldahlgren) [inline var/spack/repos/builtin/packages/gromacs/package.py:57]: Confirmed sha256.

**Comment by tldahlgren** [comment]
@marvinbernhardt Care to review this PR since you are one of its maintainers?

File: var/spack/repos/builtin/packages/gromacs/package.py [hunk 2]
```diff
@@ -135,6 +137,33 @@ def patch(self):
...
 
+        if '+cuda' in self.spec:
+            # Upstream supports building of last two major versions of Gromacs.
+            # Older versions of Gromacs need to be patched to build with more recent
+            # versions of CUDA library.
+
+            # Hardware version 3.0 is supported up to CUDA 10.2
+            # (Gromacs 4.6-2019 needs to be patched, 2020.4 is handling it correctly)
+
+            if self.spec.satisfies('@4.6:2020.3^cuda@11:'):
```
Comment (marvinbernhardt) [inline var/spack/repos/builtin/packages/gromacs/package.py:148]: comment does not match spec (:2019 vs :2020.3)

**Review by marvinbernhardt — DISMISSED** [review]
This looks all good. I knew about this issue and this is a nice way of handling them!

**Comment by tldahlgren** [comment]
> This looks all good. I knew about this issue and this is a nice way of handling them!

Thank you for your review.

File: var/spack/repos/builtin/packages/gromacs/package.py (hunk 2 above)
Comment (tldahlgren) [inline var/spack/repos/builtin/packages/gromacs/package.py]: Please update the comment to address Marvin's issue.

**Review by tldahlgren — CHANGES_REQUESTED** [review]

File: var/spack/repos/builtin/packages/gromacs/package.py (hunk 2 above)
Comment (w8jcik) [reply var/spack/repos/builtin/packages/gromacs/package.py:148]: Sorry for this, I changed the spec at some point and didn't update the comment.

The code is correct. Fixed the comment.

Comment (w8jcik) [reply var/spack/repos/builtin/packages/gromacs/package.py]: Done. Sorry for the delay, it was night at my place :)

Comment (tldahlgren) [reply var/spack/repos/builtin/packages/gromacs/package.py]: Np.  Thanks for the fix.

**Review by tldahlgren — APPROVED** [review]

**Pull request merged**